app/app.go

```diff
@@ -259,6 +262,15 @@ func New(
 	cdc := encodingConfig.Amino
 	interfaceRegistry := encodingConfig.InterfaceRegistry
 
+	// set the protocol version to a specific value if specified. This is used
+	// for testing purposes.
```

Comment:
[Question & Optional Suggestion] What is the impact of this if not used for testing purposes? Couldn't this alternatively be a function in the testground package?

Reply:
the app version must be set internally, so unfortnuately we can't do that since we don't have direct access to the baseapp, only the interface that wraps around it. However, we should be able to remove this now that we have the latest version of our fork of the sdk, since that version sets the appversion internally similar to this

to answer the question tho, it could be used to change the app version. The app version is included in and checked when verifying the header, so doing this without coordination with the rest of the network would result in the node halting.